src/7B440.c

```diff
@@ -71,6 +172,88 @@ void input_to_move_vector(f32* angle, f32* magnitude) {
     *magnitude = mag;
 }
 
-INCLUDE_ASM(s32, "7B440", func_800E23FC);
+void func_800E23FC(f32* arg0, f32* arg1) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 stickX = gGameStatusPtr->stickX;
+    f32 stickY = -gGameStatusPtr->stickY;
+    f32 tempMax = 70.0f;
+    f32 temp1;
+    f32 temp2;
+
+    temp1 = dist2D(0.0f, 0.0f, stickX, stickY);
+    if (temp1 >= tempMax) {
+        temp1 = tempMax;
+    }
+
+    temp2 = clamp_angle(atan2(0.0f, 0.0f, stickX, stickY) + gCameras[0].currentYaw);
+    if (temp1 == 0.0f) {
+        temp2 = playerStatus->targetYaw;
+    }
+
+    *arg0 = temp2;
+    *arg1 = temp1;
+}
 
 INCLUDE_ASM(s32, "7B440", func_800E24F8);
+// void func_800E24F8(void) {
```

Comment:
Throwing your comment on my PR back at ya: if this is close, wrap it in nonmatching 😉